main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
No, what I don't like is mostly the line end normalization here.

Especially, I just tested and `fgets()` from the my glibc does put both the `\r` and `\n` in the returned string, and I truly want MIO to keep having the same behavior whether run on memory or file backend.

Results of my tests on glibc:
* LF is put in the string
* CR/LF is put in the string
* CR is not handled specially.

So the current MIO code before these changes present the same behavior as glibc on GNU/Linux, so it's good.  Don't change that :)